pkg/generators/clientgen/gen.go

```diff
@@ -376,7 +397,117 @@ func (g *Generator) generateSubInterfaces(ctx *genall.GenerationContext) error {
 					return
 				}
 
-				a, err := internal.NewAPI(root, info, string(version.Version), gv.PackageName, !isClusterScoped(info), hasStatusSubresource(info), &outContent)
+				/*
+					This is the current list of unique genclient markers as of 1.23
+
+					// +genclient
+					// +genclient:method=ApplyScale,verb=apply,subresource=scale,input=Scale,result=Scale
+					// +genclient:method=ApplyScale,verb=apply,subresource=scale,input=k8s.io/api/autoscaling/v1.Scale,result=k8s.io/api/autoscaling/v1.Scale
+					// +genclient:method=CreateScale,verb=create,subresource=scale,input=k8s.io/api/autoscaling/v1.Scale,result=k8s.io/api/autoscaling/v1.Scale
+					// +genclient:method=CreateToken,verb=create,subresource=token,input=k8s.io/api/authentication/v1.TokenRequest,result=k8s.io/api/authentication/v1.TokenRequest
+					// +genclient:method=GetScale,verb=get,subresource=scale,result=Scale
+					// +genclient:method=GetScale,verb=get,subresource=scale,result=k8s.io/api/autoscaling/v1.Scale
+					// +genclient:method=UpdateApproval,verb=update,subresource=approval,input=k8s.io/api/certificates/v1.CertificateSigningRequest,result=k8s.io/api/certificates/v1.CertificateSigningRequest
+					// +genclient:method=UpdateEphemeralContainers,verb=update,subresource=ephemeralcontainers
+					// +genclient:method=UpdateScale,verb=update,subresource=scale,input=Scale,result=Scale
+					// +genclient:method=UpdateScale,verb=update,subresource=scale,input=k8s.io/api/autoscaling/v1.Scale,result=k8s.io/api/autoscaling/v1.Scale
+					// +genclient:noStatus
+					// +genclient:nonNamespaced
+					// +genclient:noVerbs
+					// +genclient:onlyVerbs=create
+					// +genclient:onlyVerbs=create,delete
+					// +genclient:readonly
+					// +genclient:skipVerbs=deleteCollection
+					// +genclient:skipVerbs=get,update
+				*/
+
+				var additionalMethods []internal.AdditionalMethod
+				var skipVerbs []string
+				var onlyVerbs []string
+
+				genclientMarkers := info.Markers[genclientMarker.Name]
+
+				namespaceScoped := info.Markers.Get(nonNamespacedMarker.Name) == nil
+				noVerbs := info.Markers.Get(noVerbsMarker.Name) != nil
+				hasStatus := hasStatusSubresource(info)
```

Comment:
That's because it does the additional task of checking whether the resource field is present for the type or not - https://github.com/kcp-dev/code-generator/blob/4977dcfd75f4ea485070682f01db1fc851ceb666/pkg/generators/clientgen/gen.go#L452-L457, in addition to checking if the marker exists.

Reply:
Do we emit some sort of validation error when a user asks for a status update but their type does not have it?

Reply:
Nope, that condition doesn't arise. Because this is a `noStatus` marker. If 
1. `noStatus` marker is present and status subfield is present or not present - we don't scaffold status sub resources.
2. `noStatus` marker is not present - if `Status` field is there its scaffolded, if not its not scaffolded.